### PR TITLE
Add support for more args for `mmseqs` clustering

### PR DESCRIPTION
**Pull request by SagiPolaczek**
# ✅ 

Added additional arguments for `mmseqs` clustering + fixing typos

### CODE REVIEW

**Comment by mosheraboh** [comment]
@YoelShoshan , can you please review it?

File: fusedrug/utils/cluster/cluster_using_mmseqs.py [hunk 1]
```diff
@@ -87,15 +90,17 @@ def cluster(
     *,
     input_fasta_filename: str,
     output_dir: str,
-    cluster_min_seqeunce_identity: float = 0.4,
-    threads: int = 30,
+    cluster_min_sequence_identity: float = 0.4,
+    threads: Optional[int] = None,
     cluster_method: str = "cluster",
     deduplicate: bool = True,
+    override_workspace: bool = False,
```
Comment (YoelShoshan) [inline fusedrug/utils/cluster/cluster_using_mmseqs.py:97]: please add it on the docstring

File: fusedrug/utils/cluster/cluster_using_mmseqs.py [hunk 2]
```diff
@@ -111,14 +116,17 @@ def cluster(
             cmd = f"gunzip {input_fasta_filename}.gz"
             _run_system_cmd(cmd)
 
-    os.makedirs(join(output_dir, "mmseqs_workspace"))
+    # Create workspace (supports overriding)
+    workspace_dir = join(output_dir, "mmseqs_workspace")
+    if override_workspace and os.path.exists(workspace_dir):
+        shutil.rmtree(workspace_dir)
 
+    os.makedirs(workspace_dir)
```
Comment (YoelShoshan) [inline fusedrug/utils/cluster/cluster_using_mmseqs.py:124]: great - I like this (not setting exist_ok=True) - as we don't want to clutter and make people unintentionally delete directories with other things.

File: fusedrug/utils/cluster/cluster_using_mmseqs.py [hunk 3]
```diff
@@ -222,14 +239,31 @@ def cluster(
...
+    res = subprocess.run(cmd, shell=True, check=False, capture_output=capture_output)
+    if res.stdout and len(res.stdout) > 0:
         print("stdout=")
         print(res.stdout.decode())
-    if len(res.stderr) > 0:
+    if res.stderr and len(res.stderr) > 0:
         print("stderr=")
         print(res.stderr.decode())
-    if res.returncode != 0:
+    if res.returncode and res.returncode != 0:
```
Comment (YoelShoshan) [inline fusedrug/utils/cluster/cluster_using_mmseqs.py:251]: can you describe this line modification motivation ?

**Review by YoelShoshan — CHANGES_REQUESTED** [review]
Great PR !
Awesome to see that you found important and useful flags - especially the one controlling the memory requirements :)

See comments inline, not approving yet just due to 
line 251: "if res.returncode and res.returncode != 0:"
I'd like to understand it a bit better before approving.

File: fusedrug/utils/cluster/cluster_using_mmseqs.py (hunk 1 above)
Comment (SagiPolaczek) [reply fusedrug/utils/cluster/cluster_using_mmseqs.py:97]: My bad,
Done 👍

File: fusedrug/utils/cluster/cluster_using_mmseqs.py (hunk 3 above)
Comment (SagiPolaczek) [reply fusedrug/utils/cluster/cluster_using_mmseqs.py:251]: Just for future reference:

Double checked that running twice the script :
```python
import subprocess

res = subprocess.run("echo 1", shell=True, check=False, capture_output=?)

print(res.stdout)
print(res.stderr)
print(res.returncode)
```

With `capture_output=True` we get:
```
b'1\n'
b''
0
```

And `capture_output=False` gives us:
```
1
None
None
0
```

**Review by YoelShoshan — APPROVED** [review]
Looks great!
Approved.

**Pull request merged**